specs/local-context-spec.md

```diff
@@ -0,0 +1,274 @@
+- **AI working directory**: the directory in which the AI system is preparing to or is actively working within. If an AI system supports having a mounted directory (like an IDE opening a coding folder), this mounted directory is the default AI working directory. If the AI system's mounted directory is `~/myproject/`, then the AI working directory is by default `~/myproject/`. If the AI system determines that it needs to add/edit a file within `nested/paths/`, the AI system's AI working directory changes to be `~/myproject/nested/paths/` during that task. The mounted directory stays the same but, depending on what the AI system's agentic decisions, the AI working directory MAY change during the course of the AI system's work.
+- **Client context path**: the relative path string that is used to find context files and configurations.
+- **Context configuration**: a configuration file that provides additional rules for searching and filtering context to an AI system while it is running within AI working directory.
+- **Ancestor context**: When AI systems have a specific mounted directory (typically a project directory) and the AI working directory is at a nested directory, all context and configuration between these two directories is the ancestor context. For example, if the mounted directory is `~/myproject` and the AI working directory is `~/myproject/nested/paths/feature/` then context and context configuration that might exist within `feature`, `paths`, `nested`, `myproject` are the entirety of possible ancestor context.
+- **Glob pattern**: A string pattern that is used to match file paths. This specification uses the [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax inclusive of the `**` pattern. Glob patterns must also be treated as case insensitive.
+
+
+### Scope requirements
+
+If an AI system supports client-hosted context, it MUST support one or more of the following context scopes:
+
+- `global context`: context available to the AI system across the client environment.
+- `static directory context`: context available to the AI system within the mounted directory (typically the root of a project) or the AI system's working directory if it does not work within a mounted directory.
+- `dynamic subdirectory context`: context available to the AI system within subdirectories of the AI system's working directory (typically subdirectories of a project). The context files and configuration are dynamically selected and merged based on discovering context files and configuration at or between the mounted directory and the AI systems working directory.
+
+### General path requirements
+
+AI systems MUST normalize all path separators for the client operating system and expand the home directory symbol (~) using standard OS conventions.
+
+AI systems SHOULD avoid following symlinks by default to avoid unintentional context inclusion.
```

Comment:
What if `~/.context` is a symlink? Asking because I typically keep all of my config in a git repo in `~/.dotfiles` and then symlink everything from there so that all my configs are tracked in git